xmtp_mls/src/storage/errors.rs

```diff
@@ -89,7 +89,16 @@ impl RetryableError for openmls::group::CreateCommitError<sql_key_store::SqlKeyS
     fn is_retryable(&self) -> bool {
         match self {
             Self::KeyStoreError(storage) => retryable!(storage),
-            Self::KeyPackageGenerationError(generation) => retryable!(generation),
+            Self::LeafNodeUpdateError(leaf_node_update) => retryable!(leaf_node_update),
+            _ => false,
+        }
+    }
+}
+
+impl RetryableError for openmls::treesync::LeafNodeUpdateError<sql_key_store::SqlKeyStoreError> {
+    fn is_retryable(&self) -> bool {
+        match self {
+            Self::Storage(storage) => retryable!(storage),
```

Comment:
Are all storage errors retryable here?

Reply:
Not all storage errors are retryable, but most are. Connection pool is a perfect example of a retryable error.

Answering this question I noticed some newer DB error types that aren't marked as retryable, so I'm going to add them in here. 😬